boomerang/src/nicp_to_icp.rs

```diff
@@ -97,32 +102,32 @@ pub async fn try_retrieve_icp(target: Principal) -> Result<Nat, BoomerangError>
     let boomerang_id = self_canister_id();
     let subaccount = derive_subaccount_unstaking(target);
 
-    let target_account = Account {
+    let boomerang_account = Account {
         owner: boomerang_id,
         subaccount: Some(subaccount),
     };
 
-    let icp_balance_e8s: u64 = match icp_client.balance_of(target_account).await {
+    let icp_balance_e8s: u64 = match icp_client.balance_of(boomerang_account).await {
         Ok(balance) => balance.0.try_into().unwrap(),
         Err((code, message)) => {
-            return Err(BoomerangError::GenericError { code, message });
+            return Err(BoomerangError::BalanceOfError(format!(
+                "code: {code} - message: {message}"
+            )));
         }
     };
 
-    if icp_balance_e8s < TRANSFER_FEE {
-        return Err(BoomerangError::NotEnoughICP);
+    if icp_balance_e8s == 0 {
+        return Err(BoomerangError::IcpNotAvailable);
```

Comment:
Keep the precedent error, this one is confusing, what does it mean ICP is not available? Will it become available?

Reply:
We shall disagree on this one. To me, NotEnoughICP infers that you expect to have ICP but in the end you don't. 
This error intends to tell the user that the period of 6 months is not over yet and that boomerang did not receive the funds yet. Hence IcpNotAvailable but maybe there is a better name.

Reply:
> To me, NotEnoughICP infers that you expect to have ICP but in the end you don't.

No, NotEnoughICP just means that there is not enough ICP to perform the desired action.

> This error intends to tell the user that the period of 6 months is not over yet and that boomerang did not receive the funds yet. Hence IcpNotAvailable but maybe there is a better name.

Then the error is misleading and wrong. The code is absolutely not checking for the desired information. And the name is still bad because ICPNotAvailable could mean that ICP the protocol is not available.